src/callback.rs

```diff
@@ -35,6 +35,31 @@ impl PyCallbackOutput for () {
     const ERR_VALUE: Self = ();
 }
 
+/// A helper trait to result-wrap the return values of functions and methods.
+#[doc(hidden)]
+pub trait IntoPyResult<R> {
+    fn into_py_result(self) -> R;
+}
```

Comment:
I was thinking that we'd restrict the return value to `PyResult`:

```suggestion
pub trait IntoPyResult<T> {
    fn into_py_result(self) -> PyResult<T>;
}
```

Was there a reason you preferred not to do so?